src/components/global/Header/ConnectWalletButton.tsx

```diff
@@ -118,40 +143,139 @@ const WalletButton = () => {
+              onClick={() => setModalOpen(!modalOpen)}
+              src="/images/icon-close.svg"
+              height="25px"
+              width="25px"
+              alt="close modal"
+            />
+          </TokenModal.TitleRow>
+          <TokenModal.AddressRow>
+            <TokenModal.WalletIcon>
+              <Jazzicon
+                diameter={25}
+                seed={jsNumberForAddress(walletWeb3ReactContext.account)}
+              />
+            </TokenModal.WalletIcon>
+            <TokenModal.AddressName>
+              {addressHidden
+                ? obscureAddress(walletWeb3ReactContext.account)
+                : walletWeb3ReactContext.account}
+            </TokenModal.AddressName>
+            <TokenModal.ToggleAddress
+              onClick={() => setAddressHidden(!addressHidden)}
+              src={
+                addressHidden
+                  ? '/images/icon-eye-hide.png'
+                  : '/images/icon-eye-show.png'
+              }
+              alt={addressHidden ? 'Show Address' : 'Hide Address'}
+            />
+          </TokenModal.AddressRow>
+          <TokenModal.WalletActionsRow>
+            <TokenModal.WalletAction
+              onClick={() => {
+                navigator.clipboard.writeText(walletWeb3ReactContext.account)
+                setCopiedAddress(true)
+                setTimeout(() => setCopiedAddress(false), 2000)
+              }}
+            >
+              <img
+                src="/images/icon-copy.svg"
+                height="15px"
+                width="16px"
+                alt="Copy Address"
+              />
+              {copiedAddress ? `Copied!` : `Copy Address`}
+            </TokenModal.WalletAction>
+            <NextLink
+              href={`https://etherscan.io/address/${walletWeb3ReactContext.account}`}
+            >
+              <TokenModal.WalletAction>
+                <img
+                  src="/images/icon-export.svg"
+                  height="15px"
+                  width="15px"
+                  alt="Open in Etherscan"
+                />
+                Etherscan
+              </TokenModal.WalletAction>
+            </NextLink>
+            <TokenModal.WalletAction
+              onClick={() => walletWeb3ReactContext.deactivate()}
+            >
+              <img
+                src="/images/icon-wallet.svg"
+                height="15px"
+                width="15px"
+                alt="Change Wallet"
+              />
+              Change Wallet
+            </TokenModal.WalletAction>
+          </TokenModal.WalletActionsRow>
+          <TokenModal.BigRow>
+            <TokenModal.BankTitle>Bankless DAO / $BANK</TokenModal.BankTitle>
+            <TokenModal.BigBank>
+              <strong>{whaleBalance}</strong>
+              <img
+                onMouseEnter={() => {
+                  setAnimateToken(true)
+                }}
+                onMouseLeave={() => {
+                  setAnimateToken(false)
```

Comment:
the one thing that I notice here is that it cuts the animation off midway. maybe we do an on-click enqueue start that resets when done. that way user can't click infinitely and continue to reset/stutter it

Reply:
It's using mouseOver at the moment, do you want to switch to onClick? I can add some queue logic that keeps it from running forever once that is done, I was just trying to keep it simple but I see what you mean.

Reply:
yeah, a nice to have though

Reply:
> the one thing that I notice here is that it cuts the animation off midway. maybe we do an on-click enqueue start that resets when done. that way user can't click infinitely and continue to reset/stutter it

I changed this to run the animation when you click. It also blocks a new restart of the animation while it is still running. I think I prefer the click to mouse hover.